### PR TITLE
🐛 Destination snowflake: use 200MB batches

### LINKED ISSUE #31741: Update snowflake async to use 200mb batches again
From [slack](https://airbytehq-team.slack.com/archives/C05H8UCNCK0/p1698082367311059)

* the new async framework defaults to [50 MiB](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/java/airbyte-cdk/db-destinations/src/main/java/io/airbyte/cdk/integrations/destination/staging/AsyncFlush.java#L47) batches
  * bigquery is overriding this value to 25 MiB in GCS mode, and 20% of max memory in standard inserts mode
* the previous code was using [200 MiB](https://github.com/airbytehq/airbyte/blob/556e9adb0789671d5f644bff621848513355f684/airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/destination/record_buffer/FileBuffer.java#L27), and cites snowflake documentation for that number

Rollout
* Do a few test syncs in the perf workspace to make sure that (1) it works and (2) it didn't get slower

### PR DESCRIPTION
**Pull request by edgao**
closes https://github.com/airbytehq/airbyte/issues/31741

test sync running - https://cloud.airbyte.com/workspaces/b61bc266-ef3c-460c-af2f-f70da4a2993c/connections/a54f466c-83a6-40e7-ae86-6e79961040d5/job-history#7796162::0. Verified that it now has 200MB batches:
image

Previous syncs took 500 - 800 COPY statements for the 10M faker dataset; new syncs are taking around 70 COPYs. Total sync runtime is still a bit under 1 hour.

make the factory class more useful; add a builder. Move the 50MB default into the factory class.

`new StagingConsumerFactory.Builder()` sounds dumb, I'm open to naming suggestions :shrug:

### CODE REVIEW

File: airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/java/io/airbyte/cdk/integrations/destination/staging/AsyncFlush.java [hunk 1]
```diff
@@ -37,16 +37,6 @@ class AsyncFlush implements DestinationFlushFunction {
   private final long optimalBatchSizeBytes;
   private final boolean useDestinationsV2Columns;
 
-  public AsyncFlush(final Map<StreamDescriptor, WriteConfig> streamDescToWriteConfig,
```
Comment (edgao) [inline airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/java/io/airbyte/cdk/integrations/destination/staging/AsyncFlush.java:40]: this constructor was only ever called from the overload constructor. Delete it.

File: airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeInternalStagingDestination.java [hunk 2]
```diff
@@ -165,7 +165,7 @@ public SerializedAirbyteMessageConsumer getSerializedMessageConsumer(final JsonN
           new DefaultTyperDeduper<>(sqlGenerator, snowflakeDestinationHandler, parsedCatalog, migrator, v2TableMigrator, defaultThreadCount);
     }
 
-    return new StagingConsumerFactory().createAsync(
+    return new StagingConsumerFactory.Builder(
```
Comment (gisripa) [inline airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeInternalStagingDestination.java:168]: nit: Factory pattern is to expose instance creation using a static method. `Factory.builder()...build()`

**Review by gisripa — APPROVED** [review]
minor nit (not a blocker) but overall looks good. One step towards cleaning this class 😅

File: airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeInternalStagingDestination.java (hunk 2 above)
Comment (edgao) [reply airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/SnowflakeInternalStagingDestination.java]: done in https://github.com/airbytehq/airbyte/pull/34502/commits/52a10a3959434f73b716f00a11342d135b6e36fb

**Pull request merged**